client/src/pages/Join.js

```diff
@@ -62,20 +63,35 @@ function Join() {
     const socketRef = useRef();
     socketRef.current = io.connect("/");
     const [matchId, setMatchId] = useState("");
+    const [snackbarOpen, setSnackbarOpen] = useState(false);
+    const [joinError, setJoinError] = useState("");
 
     const onTextChange = (e) => {
         setMatchId(e.target.value);
     };
 
-    const submitMatchId = (e) => {
+    const submitMatchId = async (e) => {
         e.preventDefault();
-        // TODO: call api route for joining match
-        socketRef.current.emit("join-match", { matchId });
-        socketRef.current.on("join-game-engine", (game) => {
-            setMatchState(game);
-            socketRef.current.disconnect();
-        });
-        return history.push(`/join/${matchId}`);
+        try {
+          const response = await fetch(`/api/match/join/${matchId}`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+          });
+          const data = await response.json();
+          if (response.status === '200') {
+            socketRef.current.emit('join-match', { matchId });
+            socketRef.current.on('join-game-engine', (game) => {
```

Comment:
We probably move this into `useEffect` so that it creates once per component life cycle, and when the component gets destroyed, it will remove the listener

Reply:
https://socket.io/docs/v3/listening-to-events/

.remove

Reply:
Moved the sockets to useEffect, including removing the socket listener so that the listening event is removed once the component unmounts